src/engraving/libmscore/score.cpp

```diff
@@ -535,6 +536,29 @@ void Score::fixTicks()
 
         tick += measureTicks;
     }
+
+    for (const auto& pair : spanner()) {
+        const Spanner* spannerItem = pair.second;
+        if (!spannerItem || !spannerItem->isTempoChangeRanged()) {
+            continue;
+        }
+
+        const TempoChangeRanged* tempoChange = toTempoChangeRanged(spannerItem);
+        if (!tempoChange) {
+            continue;
+        }
+
+        int tickPositionFrom = tempoChange->tick().ticks();
+        BeatsPerSecond currentBps = tempomap()->tempo(tickPositionFrom);
+        BeatsPerSecond newBps = currentBps * tempoChange->tempoChangeFactor();
+
+        std::map<int, double> tempoCurve = TConv::easingValueCurve(tempoChange->ticks().ticks(), 4, std::abs(currentBps.val - newBps.val), tempoChange->easingMethod());
```

Comment:
Change the first param to `tickPositionFrom` for clarity (and no double fetching).
Wouldn't it be better to make the hardcoded 4 here depend on the spanned duration instead? Looking at the uses for the current TempoChanges plugin there are often more markings inserted (each segment there) especially in curved ways.
So I'd make this depend on the amount of change and rate of change instead; striving for a calculation every half beat approximately.

Reply:
I know it's pretty confusing (I mean tick and ticks), but these two values are not similar. (tempoChange->ticks() vs tempoChange->tick())

So there is no double fetching and it uses a proper one

Reply:
True, read straight over it.
One of my gripes with TempoMap as well and proposal to rename ticks to len in more places